.github/workflows/prepare-transaction.yml

```diff
@@ -0,0 +1,207 @@
+  ERROR_MSG_CHAINLIST_FAILURE: >
+    **⛔️ Error:**<br>
+    Couldn't find the chain in the chainlist. Please make sure it's added to the chainlist.<br>
+    For more information on how to add a chain, please refer to the [chainlist repository](https://github.com/ethereum-lists/chains).<br>
+    :sparkles: You can edit the issue to trigger the check again. :sparkles:
+  ERROR_MSG_PREFUND_CHECK: >
+    **⛔️ Error:**<br>
+    There was an error while estimating the deployment transaction and checking the pre-fund. Please make sure that the RPC URL is valid and reachable.<br>
+    :sparkles: You can edit the issue to trigger the check again. :sparkles:
+  SUCCESS_MSG: >
+    **✅ Success:**<br>
+    The issue description is valid:<br>
+    - The RPC URL is valid<br>
+    - The chain is in the chainlist<br>
+    - The deployer address is pre-funded<br>
+    :sparkles: The team will be in touch with you soon :sparkles:
+
+jobs:
+  prepareTransaction:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Validate the issue description and get the RPC URL
+        id: validate_description
+        run: |
+          trim() {
+              # 3.5.3 Shell Parameter Expansion at https://www.gnu.org/software/bash/manual/bash.html
+              local var="$*"
+              # remove leading whitespace characters
+              var="${var#"${var%%[![:space:]]*}"}"
+              # remove trailing whitespace characters
+              var="${var%"${var##*[![:space:]]}"}"
+              echo "$var"
+          }
+
+          rpc_url=$(echo $ISSUE_BODY | egrep -o 'https?://[^ ]+' -m 1 | head -1)
+
+          if [ -z "$rpc_url" ]; then
+            echo "COMMENT_OUTPUT="$ERROR_MSG_RPC_FAILURE"" >> $GITHUB_ENV
+            exit 1
+          fi
+
+          rpc_url=$(trim $rpc_url)
+          echo "Extracted RPC url: $rpc_url. Trying to get the chain id..."
+          response=$(curl $rpc_url --location --header \
+            'Content-Type: application/json' --data '{
+                          "jsonrpc": "2.0",
+                          "method": "eth_chainId",
+                          "params": [],
+                          "id": 5413
+                      }')
+
+          if jq -e . >/dev/null 2>&1 <<< "$response"; then
+              chain_id=$(echo "$response" | jq -r '.result')
+              chain_id=${chain_id#0x}
+              chain_id=$((16#$chain_id))
+              echo "chain_id=$chain_id" >> $GITHUB_OUTPUT
+              echo "rpc_url=$rpc_url" >> $GITHUB_OUTPUT
+          else
+              echo "Failed to parse JSON, or got false/null"
+              echo "COMMENT_OUTPUT="$ERROR_MSG_RPC_FAILURE"" >> $GITHUB_ENV
+              exit 1
+          fi
+
+      - name: Check if the factory already deployed
+        if: success() && ${{ steps.validate_description.outputs.rpc_url != null }}
+        env:
+          RPC_URL: ${{ steps.validate_description.outputs.rpc_url }}
+        run: |
+          factory_code=$(curl $RPC_URL --location --header \
```

Comment:
one trick you can do in these cases is:

```bash
curl --data '@-' <<JSON
{
  "jsonrpc": "2.0",
  "method": "eth_getCode",
  "params": ["$FACTORY_ADDRESS", "latest"],
  "id": 5413
}
JSON
```

This allows you to use variable expansion directly.

(It's fine as is and just as readable; just pointing it out in case there are many variable expansions, then it is easier to do it with HEREDOCs.)